src/components/Widgets/ControlHOC.js

```diff
@@ -80,14 +81,25 @@ class ControlHOC extends Component {
   };
 
   render() {
-    const { controlComponent, field, value, metadata, onChange, onAddAsset, onRemoveAsset, getAsset } = this.props;
+    const {
+      controlComponent,
+      field,
+      value,
+      metadata,
+      onChange,
+      onAddAsset,
+      onRemoveAsset,
+      getAsset,
+      registerDependency,
+    } = this.props;
     return React.createElement(controlComponent, {
       field,
       value,
       metadata,
       onChange,
       onAddAsset,
       onRemoveAsset,
+      registerDependency,
```

Comment:
Yeah, that's a good point re: naming. See my comment below for more on the removal and updating.